new consume test fails

`test_issue981_consume.py::test1` is a new test that was created for #981. It worked in the initial PR. Now in CIs on main it's failing:
https://github.com/oceanprotocol/ocean.py/actions/runs/3182107115/jobs/5187639946#step:7:1740

Calina: "The provider is not serving it"

Some of the traceback (see url above for all):
```text
=================================== FAILURES ===================================
____________________________________ test1 _____________________________________

self = <urllib3.connectionpool.HTTPConnectionPool object at 0x7ff814285850>
conn = <urllib3.connection.HTTPConnection object at 0x7ff8145f3b20>
method = 'GET'
url = '/api/services/download?documentId=did%3Aop%3A48d9375d102abd546[65](https://github.com/oceanprotocol/ocean.py/actions/runs/3182107115/jobs/5187639946#step:7:66)0bb7d256ec2bf781b378bca385f504e0[68](https://github.com/oceanprotocol/ocean.py/actions/runs/3182107115/jobs/5187639946#step:7:69)97ad2d05373&service...fa0db909623bbbda6a7fff468e443255a3f91a9787b5731eabd811e05481db3cb8f7491039e375906f44028b02f77b0e158da604ec3159dd62311c'
timeout = Timeout(connect=3, read=3, total=None), chunked = False
httplib_request_kw = {'body': None, 'headers': {'User-Agent': 'python-requests/2.28.1', 'Accept-Encoding': 'gzip, deflate', 'Accept': '*/*', 'Connection': 'keep-alive'}}
timeout_obj = Timeout(connect=3, read=3, total=None), read_timeout = 3

    def _make_request(
        self, conn, method, url, timeout=_Default, chunked=False, **httplib_request_kw
    ):
        """
        Perform a request on a given urllib connection object taken from our
        pool.
    
        :param conn:
            a connection from one of our connection pools
    
        :param timeout:
            Socket timeout in seconds for the request. This can be a
            float or integer, which will set the same timeout value for
            the socket connect and the socket read, or an instance of
            :class:`urllib3.util.Timeout`, which gives you more fine-grained
            control over your timeouts.
        """
        self.num_requests += 1
    
        timeout_obj = self._get_timeout(timeout)
        timeout_obj.start_connect()
        conn.timeout = timeout_obj.connect_timeout
    
        # Trigger any extra validation we need to do.
        try:
            self._validate_conn(conn)
        except (SocketTimeout, BaseSSLError) as e:
            # Py2 raises this as a BaseSSLError, Py3 raises it as socket timeout.
            self._raise_timeout(err=e, url=url, timeout_value=conn.timeout)
            raise
    
        # conn.request() calls http.client.*.request, not the method in
        # urllib3.request. It also calls makefile (recv) on the socket.
        try:
            if chunked:
                conn.request_chunked(method, url, **httplib_request_kw)
            else:
                conn.request(method, url, **httplib_request_kw)
    
        # We are swallowing BrokenPipeError (errno.EPIPE) since the server is
        # legitimately able to close the connection after sending a valid response.
        # With this behaviour, the received response is still readable.
        except BrokenPipeError:
            # Python 3
            pass
        except IOError as e:
            # Python 2 and macOS/Linux
            # EPIPE and ESHUTDOWN are BrokenPipeError on Python 2, and EPROTOTYPE is needed on macOS
            # https://erickt.github.io/blog/2014/11/19/adventures-in-debugging-a-potential-osx-kernel-bug/
            if e.errno not in {
                errno.EPIPE,
                errno.ESHUTDOWN,
                errno.EPROTOTYPE,
            }:
                raise
    
        # Reset the timeout for the recv() on the socket
        read_timeout = timeout_obj.read_timeout
    
        # App Engine doesn't have a sock attr
        if getattr(conn, "sock", None):
            # In Python 3 socket.py will catch EAGAIN and return None when you
            # try and read into the file pointer created by http.client, which
            # instead raises a BadStatusLine exception. Instead of catching
            # the exception and assuming all BadStatusLine exceptions are read
            # timeouts, check for a zero timeout before making the request.
            if read_timeout == 0:
                raise ReadTimeoutError(
                    self, url, "Read timed out. (read timeout=%s)" % read_timeout
                )
            if read_timeout is Timeout.DEFAULT_TIMEOUT:
                conn.sock.settimeout(socket.getdefaulttimeout())
            else:  # None or a value
                conn.sock.settimeout(read_timeout)
    
        # Receive the response from the server
        try:
            try:
                # Python 2.7, use buffering of HTTP responses
                httplib_response = conn.getresponse(buffering=True)
            except TypeError:
                # Python 3
                try:
                    httplib_response = conn.getresponse()
                except BaseException as e:
                    # Remove the TypeError from the exception chain in
                    # Python 3 (including for exceptions like SystemExit).
                    # Otherwise it looks like a bug in the code.
>                   six.raise_from(e, None)

/opt/hostedtoolcache/Python/3.8.14/x64/lib/python3.8/site-packages/urllib3/connectionpool.py:449: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
```

--- a/tests/integration/ganache/test_issue981_consume.py
+++ b/tests/integration/ganache/test_issue981_consume.py
@@ -10,9 +10,12 @@ import shutil
 import pytest
 from web3 import Web3
 
+from ocean_lib.agreements.service_types import ServiceTypes
+from ocean_lib.data_provider.data_service_provider import DataServiceProvider
 from ocean_lib.models.datatoken import Datatoken
 from ocean_lib.ocean.ocean import Ocean
 from ocean_lib.web3_internal.wallet import Wallet
+from tests.resources.ddo_helpers import get_first_service_by_type
 
 
 @pytest.mark.integration
@@ -23,27 +26,28 @@ def test1(
     consumer_wallet: Wallet,
     tmp_path,
 ):
+    data_provider = DataServiceProvider
     ocean = Ocean(config)
 
+    # Publish
     url = "https://cexa.oceanprotocol.io/ohlc?exchange=binance&pair=ETH/USDT"
     name = "CEXA ETH-USDT"
     asset = ocean.assets.create_url_asset(name, url, publisher_wallet)
+
+    # Initialize service
+    service = get_first_service_by_type(asset, ServiceTypes.ASSET_ACCESS)
+    response = data_provider.initialize(
+        did=asset.did, service=service, consumer_address=consumer_wallet.address
+    )
+
+    # Share access
     datatoken_address = asset.datatokens[0]["address"]
     datatoken = Datatoken(ocean.web3, datatoken_address)
-
     to_address = consumer_wallet.address
     datatoken.mint(to_address, ocean.to_wei(10), publisher_wallet)
 
+    # Consume
     order_tx_id = ocean.assets.pay_for_access_service(asset, consumer_wallet)
-
-    # Commented out, we shouldn't need this
-    # Initialize service
-    # response = data_provider.initialize(
-    #    did=asset.did, service=service, consumer_address=consumer_wallet.address
-    # )
-
-    # consumer now has access. He downloads the asset.
-    # If the connection breaks, consumer can request again by showing order_tx_id.
     file_path = ocean.assets.download_asset(
         asset=asset,
         consumer_wallet=consumer_wallet,

